### PR TITLE
Creating method to set the flags ins structural solids

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR creates a method to set the flags in the structural solids. This way the confusion is avoided when using the flags for CL

Besides, I added some Doxygen comments and I removed one duplicated method from the UL and TL

### CODE REVIEW

**Review by RiccardoRossi — COMMENTED** [review]
@loumalouomega  there are times when you want to compute stress and C, others in which you just want stress and finally cases in which you don't want any of those. I would not hide this feature...

**Comment by loumalouomega** [comment]
It is not hide, there are default parameters, check it out

**Comment by loumalouomega** [comment]
imagen

**Comment by RiccardoRossi** [comment]
I don't think adding this is an improvment. It only makes the mechanism more opaque... let's see what others think though

**Comment by loumalouomega** [comment]
I am adding precisely to avoid confusions with the small displacement element and the large strain elements

**Comment by RiccardoRossi** [comment]
as i am telling i vote against this... let's see if others back you up ...

**Comment by loumalouomega** [comment]
OK

**Comment by loumalouomega** [comment]
There is another reason, is for the computations done in the base_solid_element, to take into account the special behaviour of the small_displacement_element

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp [hunk 1]
```diff
@@ -144,6 +142,20 @@ ConstitutiveLaw::StressMeasure BaseSolidElement::GetStressMeasure() const
 /***********************************************************************************/
 /***********************************************************************************/
 
+void BaseSolidElement::SetConstituveLawFlags(
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp]: I would call this function `SetDefaultConstituveLawFlags`, maybe this makes it more clear

File: applications/StructuralMechanicsApplication/custom_elements/small_displacement.h [hunk 2]
```diff
@@ -138,6 +142,18 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) SmallDisplacement
     {
     }
 
+    /**
+     * @brief Sets the flags of the constitutive law
+     * @param rConstituveLawFlags The flags of the constitutive law
+     * @param ComputeStress If the stress vector is computed or not
+     * @param ComputeConstitutiveTensor If the constitutive tensor is computed or not
+     */
+    void SetConstituveLawFlags(
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/small_displacement.h]: see name change suggestion

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp [hunk 3]
```diff
@@ -667,9 +675,7 @@ void BaseSolidElement::CalculateOnIntegrationPoints(
         
         // Set constitutive law flags:
         Flags &ConstitutiveLawOptions=Values.GetOptions();
-        ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
-        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, false);
-        ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+        SetConstituveLawFlags(ConstitutiveLawOptions);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp]: I think the arguments are wrong here, shouldn't it be `SetConstituveLawFlags(ConstitutiveLawOptions, false);` ?

**Review by philbucher — COMMENTED** [review]
In principle I like the idea but there are some comments
Having default values makes sense, otherwise some copy-paste errors are likely to happen

**Comment by loumalouomega** [comment]
Right, I will take in consideration your suggestions

**Comment by loumalouomega** [comment]
I change the method to accommodate the suggestions of @RiccardoRossi, and make him happy . Now instead of a method to set the flags I just create a method used to know if the element provides the strain or not, It is false in the base solid element and true for the small displacement

**Review by RiccardoRossi — COMMENTED** [review]
this looks better to me, however....you are not returning a true anywhere, and this should happen for the kinematic_linear_element as a minimum

File: applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp [hunk 4]
```diff
@@ -57,6 +57,14 @@ SmallDisplacement::~SmallDisplacement()
 /***********************************************************************************/
 /***********************************************************************************/
 
+bool SmallDisplacement::UseElementProvidedStrain()
+{
+    return true;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp:62]: @RiccardoRossi he does here

**Review by philbucher — COMMENTED** [review]
Looks good to me!

File: applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp (hunk 4 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp:62]: Thanks @philbucher

**Review by RiccardoRossi — APPROVED** [review]
ok, my bad. approving

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**